src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -211,152 +207,80 @@ public bool TryGetTelemetryId(out Guid telemetryId)
+                    suggestedActions.Add(ConvertToSuggestedAction(action));
                 }
 
-                return actions.Count == 0
-                    ? null
-                    : new SuggestedActionSet(set.CategoryName, actions.ToImmutable(), set.Title, set.Priority, set.ApplicableToSpan);
-            }
-
-            private ImmutableArray<SuggestedActionSet> InlineActionSetsIfDesirable(ImmutableArray<SuggestedActionSet> allActionSets)
-            {
-                // If we only have a single set of items, and that set only has three max suggestion 
-                // offered.  Then we can consider inlining any nested actions into the top level list.
-                // (but we only do this if the parent of the nested actions isn't invokable itself).
-                if (allActionSets.Sum(a => a.Actions.Count()) > 3)
+                Span? applicableToSpan = null;
+                if (unifiedSuggestedActionSet.ApplicableToSpan.HasValue)
                 {
-                    return allActionSets;
+                    applicableToSpan = unifiedSuggestedActionSet.ApplicableToSpan.Value.ToSpan();
                 }
 
-                return allActionSets.SelectAsArray(InlineActions);
-            }
-
-            private SuggestedActionSet InlineActions(SuggestedActionSet actionSet)
-            {
-                using var newActionsDisposer = ArrayBuilder<ISuggestedAction>.GetInstance(out var newActions);
-                foreach (var action in actionSet.Actions)
-                {
-                    var actionWithNestedActions = action as SuggestedActionWithNestedActions;
+                return new SuggestedActionSet(
+                    categoryName: unifiedSuggestedActionSet.CategoryName,
+                    actions: suggestedActions,
+                    title: unifiedSuggestedActionSet.Title,
+                    priority: ConvertToSuggestedActionSetPriority(unifiedSuggestedActionSet.Priority),
+                    applicableToSpan: applicableToSpan);
 
-                    // Only inline if the underlying code action allows it.
-                    if (actionWithNestedActions?.CodeAction.IsInlinable == true)
+                // Local functions
+                ISuggestedAction ConvertToSuggestedAction(IUnifiedSuggestedAction unifiedSuggestedAction)
+                    => unifiedSuggestedAction switch
                     {
-                        newActions.AddRange(actionWithNestedActions.NestedActionSets.SelectMany(set => set.Actions));
-                    }
-                    else
+                        UnifiedCodeFixSuggestedAction codeFixAction => new CodeFixSuggestedAction(
+                            ThreadingContext, _owner, codeFixAction.Workspace, _subjectBuffer,
+                            codeFixAction.CodeFix, codeFixAction.Provider, codeFixAction.CodeAction,
+                            codeFixAction.FixAllFlavors == null ? null : ConvertToSuggestedActionSet(codeFixAction.FixAllFlavors)),
```

Comment:
would prefer if ConvertToSuggestedActionSet just could take in and return null itself.